### ISSUE TITLE
** Why is there no pull request for this feature? **

### ISSUE THREAD

**Issue by pkcakeout** [pkcakeout/pushgateway#1]
I now received multiple questions about making a pull request for this feature over the last two weeks. 

It think that this features is important and the original developers overlooked an important usecase for this feature: pushing data through a firewall which in a strict one-directional way. We need this at work since our department operates from a hospital where inbound connections to internal hardware are forbidden. Still, we want to use prometheus because it offers us a nice platform to accumulate metrics about internal services and external services to put them on a dashboard that can be reached from everywhere.

This is not a new idea, though. It has been suggested multiple times on the original repository, see:

https://github.com/prometheus/pushgateway/issues/19
https://github.com/prometheus/pushgateway/issues/83

The developers clearly state that they see this feature as an "anti pattern" to their design concepts for prometheus and therefore do not want to support/implement it.

-----

For this reason, the way I wrote the extension in a okayish, but when it comes to the parser, quite hackish way: The actual parser for data sent to the pushgateway is defined in a different project altogether and to modify that one needs to do more complex modifications. Instead , I preprocess the input that is served to the parser and while doing so, completely butcher any line-numbers that are generated for error messages generated. See [here](https://github.com/pkcakeout/pushgateway/blob/master/handler/push.go#L120). The remaining things regarding the way data is stored and deletes should be relatively solid though.

Long story short, this changeset is not good enough for creating a pull request. And as long as the original developers do not indicate that this is a useful contribution, I will leave it at the current (somewhat) hacky stage, since it allows easier merging of the change sets from the original pushgateway repository. the current changeset does not touch too many files.

If somebody wants to start using this though, I could create a docker container for this version though, which makes deployment somewhat easier? Monthly merges with the main branch would also be possible, though other people could perhaps help me with that!

**Comment by eloo** [comment]
I really like to use this feature and tested it already in a short manner.
If a pull request doesn't seem to happen in near future I would like to use docker container from your repository.

TL;DR: 
+1 for your feature
+1 for a docker container

**Comment by MrApplejuice** [comment]
The docker container has been added! Have fun with it (see the README for details).

**Comment by eloo** [comment]
Thanks for this creation of the docker container.. is already under test here ;)
